ui/src/ducks/reducer.js

```diff
@@ -29,4 +38,26 @@ const rootReducer = combineReducers({
   oidc: oidcReducer,
 });
 
+export type RootState = {
+  config: ConfigState,
+  login: LoginState,
+  oidc?: UserState,
+  app: {
+    nodes: NodesState,
+    pods: PodsState,
+    notifications: NotificationsState,
+    volumes: VolumesState,
+    solutions: SolutionsState,
+    layout: LayoutState,
+    salt: SaltState,
+    monitoring: MonitoringState,
+    alerts: AlertsState
+  }
+}
+
+export const useTypedSelector: <TSelected>(
+  selector: (state: RootState) => TSelected,
+  equalityFn?: (left: TSelected, right: TSelected) => boolean
+) => TSelected = useSelector;
```

Comment:
Two comments:

- I think we have defined custom hooks in `src/services/utils.js`(though granted, it doesn't make much sense either) - IMHO, adding a plain `src/hooks.js` could make it easier to find in the future
- the purpose of this custom hook is not obvious at first glance, so I'd suggest we add a comment to explain it (and remind the reader that they should use it instead of the standard `useSelector` hook)